code/obj/artifacts/artifact_items/dimensional_key.dm

```diff
@@ -0,0 +1,402 @@
+#define NORTH_ENTRANCE 1
+#define EAST_ENTRANCE 2
+#define SOUTH_ENTRANCE 3
+#define WEST_ENTRANCE 4
+
+/obj/item/artifact/dimensional_key
+	name = "artifact key"
+	icon = 'icons/obj/artifacts/artifactsitemS.dmi'
+	associated_datum = /datum/artifact/dimensional_key
+	var/datum/allocated_region/fissure_region
+
+	// var names here are in reference to inside the fissure - south means south entrance of the room
+	var/list/south_dummies = list()
+	var/list/north_dummies = list()
+	var/list/east_dummies = list()
+	var/list/west_dummies = list()
+
+	var/south_entr_placed = FALSE
+	var/north_entr_placed = FALSE
+	var/east_entr_placed = FALSE
+	var/west_entr_placed = FALSE
+
+	afterattack(atom/target, mob/user, reach)
+		..()
+		if (!src.artifact.activated)
+			return
+		if (!reach)
+			return
+		if (!iswall(target))
+			return
+		if (BOUNDS_DIST(user, target) > 0 || istype(get_area(target), /area/artifact_fissure) || user.z == Z_LEVEL_SECRET || user.z == Z_LEVEL_ADVENTURE)
```

Comment:
might want to check that its not `Z_LEVEL_DYNAMIC` also

Reply:
Or, check for restricted Z instead of hardcoding it